onnxruntime/core/framework/session_state.cc

```diff
@@ -113,26 +113,42 @@ void SessionState::CreateGraphInfo() {
   for (const auto& output : graph_viewer_->GetOutputs()) {
     if (output->Exists()) {
       idx = ort_value_name_idx_map_.Add(output->Name());
-      VLOGS(logger_, 1) << "Added graph output with name: " << output->Name() << " to OrtValueIndex with index: " << idx;
+      VLOGS(logger_, 1) << "Added graph output with name: " << output->Name()
+                        << " to OrtValueIndex with index: " << idx;
     }
   }
 
   LOGS(logger_, VERBOSE) << "Done saving OrtValue mappings.";
 }
 
 #if !defined(ORT_MINIMAL_BUILD)
-Status SessionState::PopulateKernelCreateInfo(KernelRegistryManager& kernel_registry_manager) {
+Status SessionState::PopulateKernelCreateInfo(KernelRegistryManager& kernel_registry_manager,
+                                              bool saving_ort_format) {
   for (auto& node : graph_.Nodes()) {
     const KernelCreateInfo* kci = nullptr;
-    ORT_RETURN_IF_ERROR(kernel_registry_manager.SearchKernelRegistry(node, &kci));
+
+    auto status = kernel_registry_manager.SearchKernelRegistry(node, &kci);
+    if (!status.IsOK() && saving_ort_format) {
+      // if we didn't find the kernel and are saving to ORT format an EP that compiles nodes is enabled.
```

Comment:
Maybe and Node is not using CPU? If it's using CPU the search below will still fail, but why search twice?

Reply:
During partitioning, all nodes will generally be assigned to valid EPs. The only exception is when we assign them to a compiling EP but do not compile them. We should only ever satisfy the 'if' when that occurs, and the only option we have currently is to fallback to CPU for those nodes. 

So the second call to SearchKernelRegistry should never fail. 

---
In reply to: [522558208](https://github.com/microsoft/onnxruntime/pull/5776#discussion_r522558208) [](ancestors = 522558208)